test/UnitTests/Microsoft.Testing.Platform.UnitTests/CommandLine/PlatformCommandLineProviderTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+
+using Microsoft.Testing.Internal.Framework;
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions;
+using Microsoft.Testing.Platform.Extensions.CommandLine;
+using Microsoft.Testing.Platform.Resources;
+using Microsoft.Testing.TestInfrastructure;
+
+using Moq;
+
+namespace Microsoft.Testing.Platform.UnitTests.CommandLine;
+
+[TestGroup]
+public class PlatformCommandLineProviderTests : TestBase
+{
+    public PlatformCommandLineProviderTests(ITestExecutionContext testExecutionContext)
+        : base(testExecutionContext)
+    {
+    }
+
+    [Arguments("Trace", true)]
+    [Arguments("Debug", true)]
+    [Arguments("Information", true)]
+    [Arguments("Warning", true)]
+    [Arguments("Error", true)]
+    [Arguments("Critical", true)]
+    [Arguments("invalid", false)]
+    public async Task IsValid_If_Verbosity_Has_CorrectValue(string dumpType, bool isValid)
+    {
+        var provider = new PlatformCommandLineProvider();
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.DiagnosticVerbosityOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [dumpType]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
```

Comment:
Could be worth validating there is no error message when it's valid.

Also, but that's up to you, I would split the valid and invalid cases as we assert differently.

Reply:
split into valid and invalid,